server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1203,6 +1236,29 @@ public Builder setTable(RelationName relationName,
                 tableVersion
             );
         }
+
+        public Builder addIndexUUIDs(RelationMetadata.Table table,
+                                     LongSupplier oidSupplier,
+                                     List<String> indexUUIDs) {
+            LinkedHashSet<String> newIndexUUIDs = new LinkedHashSet<>(table.indexUUIDs());
+            newIndexUUIDs.addAll(indexUUIDs);
+            setTable(
+                oidSupplier,
+                table.name(),
+                table.columns(),
```

Comment:
```suggestion
        public Builder addIndexUUIDs(RelationMetadata.Table table,
                                     List<String> indexUUIDs) {
            LinkedHashSet<String> newIndexUUIDs = new LinkedHashSet<>(table.indexUUIDs());
            newIndexUUIDs.addAll(indexUUIDs);
            setTable(
                table.name(),
                table.columns(),
```

See `setTable` method docs regarding oidSupplier.


Also: Is the ordering for the indexUUIDs needed?